htop.c

```diff
@@ -53,6 +55,25 @@ static void printHelpFlag() {
    exit(0);
 }
 
+static bool parsePID(char *pid_str, unsigned int *result) {
+   char *endptr;
+   unsigned long tmp_result;
+
+   if (pid_str[0] == '-')
+      return false;
+
+   errno = 0;
+   tmp_result = strtoul(pid_str, &endptr, 10);
+   if (*endptr != '\0' || errno != 0)
+      return false;
+
+   if (tmp_result > UINT_MAX)
+      return false;
```

Comment:
The check of UINT_MAX isn't to make sure whether the PID is valid. In my code (see post below), I made this check also. It's to ensure that the number can be safely type-demoted to 'unsigned int' and be put in the Hashtable.

Determining the maximum PID number is platform specific, as far as I know. For Linux it's `/proc/sys/kernel/pid_max` and PID_MAX_LIMIT. But for me, I don't wan't to include a platform specific header or read a file just to check this damn limit.

http://stackoverflow.com/questions/6294133/maximum-pid-in-linux
http://unix.stackexchange.com/questions/16883/what-is-the-maximum-value-of-the-pid-of-a-process

Still, I don't like @GiedriusS 's use of `parsePID` function. It adds another "interface" that is not necessary.